src/fklearn/causal/cate_learning/meta_learners.py

```diff
@@ -264,3 +260,195 @@ def p(new_df: pd.DataFrame) -> pd.DataFrame:
 causal_s_classification_learner.__doc__ += learner_return_docstring(
     "Causal S-Learner Classifier"
 )
+
+
+def _simulate_t_learner_treatment_effect(
+    df: pd.DataFrame,
+    learners: dict,
+    treatments: list,
+    control_name: str,
+    prediction_column: str,
+) -> pd.DataFrame:
+    """
+    Simulate the treatment effect for the T-Leaner.
+    """
+    control_fcn = learners[control_name]
+    control_conversion_probability = control_fcn(df)[prediction_column].values
+
+    scored_df = df.copy()
+
+    uplift_cols = []
+    for treatment_name in treatments:
+        treatment_fcn = learners[treatment_name]
+        treatment_conversion_probability = treatment_fcn(df)[prediction_column].values
+
+        uplift_cols.append(f"treatment_{treatment_name}__uplift")
+        scored_df[uplift_cols[-1]] = (
+            treatment_conversion_probability - control_conversion_probability
+        )
+
+    scored_df["uplift"] = scored_df[uplift_cols].max(axis=1).values
+    scored_df["suggested_treatment"] = np.where(
+        scored_df["uplift"].values <= 0,
+        control_name,
+        scored_df[uplift_cols].idxmax(axis=1).values,
+    )
+    scored_df["suggested_treatment"] = (
+        scored_df["suggested_treatment"]
+        .apply(lambda x: x.replace("__uplift", ""))
+        .values
+    )
+
+    return scored_df
+
+
+def _get_model_fcn(
+    df: pd.DataFrame,
+    treatment_col: str,
+    treatment_name: str,
+    learner: Callable,
+) -> Tuple[Callable, dict, dict]:
+    """
+    Returns a function that predicts the target column from the features.
+    """
+    df = df.loc[df[treatment_col] == treatment_name].reset_index(drop=True).copy()
+
+    return learner(df)
+
+
+def _get_learners(
+    df: pd.DataFrame,
+    unique_treatments: list,
+    control_learner: Callable,
+    control_name: str,
+    treatment_learner: Callable,
+    treatment_col: str,
+) -> dict:
+    learners: Dict[str, Callable] = {}
+
+    learner_fcn, _, _ = _get_model_fcn(
+        df, treatment_col, control_name, control_learner
+    )
+    learners[control_name] = learner_fcn
+
+    for treatment_name in unique_treatments:
+        learner_fcn, _, _ = _get_model_fcn(
+            df, treatment_col, treatment_name, treatment_learner
+        )
+        learners[treatment_name] = learner_fcn
+
+    return learners
+
+
+@curry
+def causal_t_classification_learner(
+    df: pd.DataFrame,
+    treatment_col: str,
+    control_name: str,
+    prediction_column: str,
+    learner: Callable,
```

Comment:
We can use the `LearnerFnType` type here. In the S-learner we could not use it because we were modifying the curry object including a new feature (T). I think we don't need to modifying the curry here 😄 

```suggestion
    learner: LearnerFnType,
```